bin/automatic_review.py

```diff
@@ -137,36 +161,70 @@ def patch_entry(self):
         else:
             r = 'pass'
 
-        payload = {'reviewed': r}
+        report = {'reviewed': r}
         if failing_metrics:
-            payload[ELEMENT_REVIEW_COMMENTS] = 'failed due to ' + ', '.join(failing_metrics)
-
-        rest_communication.patch_entries('samples', payload=payload, where={'sample_id': self.sample_name})
-
-
-def review_runs():
-    unreviewed_runs = rest_communication.get_documents(
-        'aggregate/all_runs',
-        paginate=False,
-        match={'proc_status': 'finished', 'review_statuses': 'not%20reviewed'}
-    )
-    for run in unreviewed_runs:
-        r = RunReviewer(run['run_id'])
-        r.patch_entry()
-
-
-def review_samples():
-    unreviewed_samples = rest_communication.get_documents(
-        'aggregate/samples',
-        paginate=False,
-        match={'proc_status': 'finished', 'reviewed': 'not%20reviewed'}
-    )
-    for sample in unreviewed_samples:
-        species = clarity.get_species_from_sample(sample['sample_id'])
-        sample_cfg = sample_config(sample, species)
-        if sample_cfg:
-            s = SampleReviewer(sample, sample_cfg, species)
-            s.patch_entry()
+            report[ELEMENT_REVIEW_COMMENTS] = 'failed due to ' + ', '.join(failing_metrics)
+
+        return report
+
+    def report(self):
+        s = '%s: %s' % (self.sample_name, self._summary)
+        self.info(s)
+        return s
+
+    def push_review(self):
+        if self.cfg:
+            rest_communication.patch_entry('samples', self._summary, 'sample_id', self.sample_name)
+
+
+def main():
+    args = _parse_args()
+    load_config()
+    log_cfg.add_handler(logging.StreamHandler(stream=sys.stdout))
+    if args.debug:
+        log_cfg.set_log_level(logging.DEBUG)
+
+    if args.run:
+        unreviewed_data = rest_communication.get_documents(
+            'aggregate/all_runs',
+            paginate=False,
+            match={'proc_status': 'finished', 'review_statuses': 'not reviewed'}
+        )
+        cls = RunReviewer
+    elif args.sample:
+        unreviewed_data = rest_communication.get_documents(
+            'aggregate/samples',
+            paginate=False,
+            match={'proc_status': 'finished', 'reviewed': 'not reviewed'}
+        )
+        cls = SampleReviewer
+    else:
+        return 1
+
+    reviewers = []
+    for d in unreviewed_data:
+        reviewer = cls(d)
+        reviewers.append(reviewer)
+
+    if args.dry_run:
```

Comment:
Reviewers now implement a `_summary` method decoupled from the Rest API push, so now we have the option to just print what will be reviewed.